src/plugins/intel_cpu/src/emitters/tpp/common/kernel_executors/brgemm.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "common/utils.hpp"
+#include "emitters/snippets/brgemm_generic.hpp"
+#include "emitters/utils.hpp"
+#include "libxsmm.h"
+
+namespace ov::intel_cpu::tpp {
+
+struct BrgemmKernelConfig : public BrgemmGenericKernelConfig {
+public:
+    BrgemmKernelConfig(const element::Type& in0_dtype, const element::Type& in1_dtype);
+    BrgemmKernelConfig() = delete;
+
+    std::unique_ptr<snippets::KernelExecutorBase::GenericConfig> get_clone_ptr() const override {
+        return std::unique_ptr<BrgemmKernelConfig>(new BrgemmKernelConfig(*this));
+    }
+
+    bool operator==(const BrgemmKernelConfig& rhs) const;
+    bool operator!=(const BrgemmKernelConfig& rhs) const {
+        return !(*this == rhs);
+    }
+
+    void update(int64_t M, int64_t N, int64_t K, int64_t LDA, int64_t LDB, int64_t LDC, float beta) override;
+
+    size_t hash() const override {
+        return m_hash;
+    }
+    size_t compute_hash() const;
+
+    libxsmm_bitfield get_compile_flags() const {
+        return m_compile_flags;
+    }
+    void set_compile_flags_with_zero_beta(bool zero_beta) {
+        if (zero_beta) {
+            m_compile_flags = m_static_params->m_compile_flags | LIBXSMM_GEMM_FLAG_BETA_0;
+        } else {
+            m_compile_flags = m_static_params->m_compile_flags;
+        }
+    }
+    bool get_prefetching_flags() const {
+        return m_static_params->m_prefetching_flags;
+    }
+    libxsmm_datatype get_type_in0() const {
+        return m_static_params->m_type_in0;
+    }
+    libxsmm_datatype get_type_in1() const {
+        return m_static_params->m_type_in1;
+    }
+    libxsmm_datatype get_type_out0() const {
+        return m_static_params->m_type_out0;
+    }
+    libxsmm_datatype get_type_exec() const {
+        return m_static_params->m_type_exec;
+    }
+#ifdef SNIPPETS_DEBUG_CAPS
+    std::string to_string() const override;
+#endif
+
+private:
+    struct StaticParams {
+        StaticParams(const element::Type& in0_dtype, const element::Type& in1_dtype);
+        virtual ~StaticParams() = default;
```

Comment:
Minor: it looks like we don't need virtual destructor here

Reply:
You are right, no child class for this and virtual destructor is not needed. Done, thanks Vladislav!